docs/Users/Tutorials/Linter_Bears_Advanced.rst

```diff
@@ -0,0 +1,134 @@
+Linter Bears - Advanced Feature Reference
+=========================================
+
+Often linters are no easy tools. To squeeze out the last bit of functionality
+and efficiency, ``@linter`` provides some advanced features ready for use.
+
+Supplying Configuration Files with ``generate_config``
+------------------------------------------------------
+
+Sometimes tools require a configuration file to run. ``@linter`` supports that
+easily with overriding ``generate_config()``.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        @staticmethod
+        def generate_config(filename, file):
+            config_file = ("value1 = 1\n"
+                           "value=2 = 2")
+            return config_file
+
+The string returned from this method is written into a temporary file before
+invoking ``create_arguments()``. If you return ``None``, no config-file is
+generated.
+
+The path of the temporary configuration-file can be accessed inside
+``create_arguments()`` via the ``config_file`` parameter:
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        @staticmethod
+        def generate_config(filename, file):
+            config_file = ("value1 = 1\n"
+                           "value=2 = 2")
+            return config_file
+
+        @staticmethod
+        def create_arguments(filename, file, config_file):
+            return "--use-config", config_file
+
+.. note::
+
+    By default no configuration file is generated.
```

Comment:
some places its ``configuration file``, others is ``configuration-file``, and L24 its config-file. maintain consistency, i vote for the first one, with a space :)